packages/edit-site/src/components/global-styles/hooks.js

```diff
@@ -39,6 +79,16 @@ function useGlobalStylesUserConfig() {
 		let parsedConfig;
 		try {
 			parsedConfig = content ? JSON.parse( content ) : {};
+			// It is very important to verify if the flag isGlobalStylesUserThemeJSON is true.
+			// If it is not true the content was not escaped and is not safe.
+			if ( ! parsedConfig.isGlobalStylesUserThemeJSON ) {
```

Comment:
I don't understand this? How can we retrieve an unsafe content at this point? Should we escape in the backend instead?

Reply:
The WordPress escape filters don't provide information about which post type we are escaping etc, they just pass content to the filter, so we rely on the flag isGlobalStylesUserThemeJSON being present to know if we should escape the json or not. An attacker can easily bypass our escaping mechanism by not including the flag isGlobalStylesUserThemeJSON so we need on every usage to verify if the flag is present otherwise things may be unsafe.
If we have a specific endpoint that verification should be done there right now we don't have one so we need to keep this check on the client.

Reply:
It's still unclear to me, why not always escape?

Reply:
> If we have a specific endpoint that verification should be done there right now we don't have one so we need to keep this check on the client.

I guess that's the missing piece for me, let's try to bring that endpoint after this PR. it seems it would solve a lot of complexity/issues.

Reply:
> It's still unclear to me, why not always escape?

Because you just receive a string of content, you can not escape a given JSON or HTML or anything in the post content with the theme.json rules. That content may be a JSON from a plugin following totally different sets of rules for what should be escaped. The escape functions don't receive anything besides a string of content that flag indicates that content is a user theme.json.